snuba/datasets/functions_processor.py

```diff
@@ -0,0 +1,56 @@
+import uuid
+from datetime import datetime
+from typing import Any, Mapping, Optional
+
+from snuba.consumers.types import KafkaMessageMetadata
+from snuba.processor import InsertBatch, MessageProcessor, ProcessedMessage
+
+
+class FunctionsMessageProcessor(MessageProcessor):
+    def process_message(
+        self, message: Mapping[str, Any], metadata: KafkaMessageMetadata
+    ) -> Optional[ProcessedMessage]:
+        functions = {}
+
+        for thread, root_frames in message["call_trees"].items():
```

Comment:
Is there a cap on the number of `root_frames` in a `call_tree`? And a cap on the children frames? This function is going to get slow very quickly if there isn't some reasonable cap.